OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -330,6 +330,43 @@ public void SetLineWidth(float width)
 			ErrorHandler.CheckGlError();
 		}
 
+		public Bitmap TakeScreenshot()
+		{
+			var rect = new Rectangle(Point.Empty, size);
+			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var data = bitmap.LockBits(rect,
+				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+
+			try
```

Comment:
I'm not saying don't ever unlock the data. I'm saying no need to unlock it if the exception is not going to be handled in a meaningful way anywhere (apart from just error logging and terminating afterwards anyways). And if it is going to be handled meaningfully somewhere, I don't understand why you would want to do this upstream in the call stack? Shouldn't normal recovery from an exception only make sense locally in a function that knows the context?